x/mongo/driver/list_collections_batch_cursor.go

```diff
@@ -11,12 +11,15 @@ import (
 	"errors"
 	"io"
 	"strings"
+	"time"
 
 	"go.mongodb.org/mongo-driver/x/bsonx/bsoncore"
 )
 
 // ListCollectionsBatchCursor is a special batch cursor returned from ListCollections that properly
 // handles current and legacy ListCollections operations.
+//
+// TODO(GODRIVER-2861): Merge ListCollectionsBatchCursor into BatchCursor.
```

Comment:
Optional: This is an exported symbol, so this TODO will show up in the pkgsite docs. It's an experimental package, so it's not a big deal, but consider moving that TODO to somewhere outside of the `ListCollectionsBatchCursor` doc comment.